server/src/main/java/io/druid/segment/realtime/plumber/RealtimePlumber.java

```diff
@@ -352,9 +390,13 @@ public void doRun()
           {
             try {
               for (Pair<FireHydrant, Interval> pair : indexesToPersist) {
-                metrics.incrementRowOutputCount(persistHydrant(pair.lhs, schema, pair.rhs));
+                metrics.incrementRowOutputCount(
+                    persistHydrant(
```

Comment:
I think it makes sense to store the metadata outside the segments in a separate file. This is because the commit metadata isn't **really** associated with an individual segment-- it's associated with a set of segments that are persisted at the same time. So storing it in the segments is asking for problems.

Sort of like this:

```json
{
  "metadata" : {...},
  "segments": [
    {"id": "datasource_2000_2001_2000_1", "hydrant": 10}, # Only hydrants through 10 are committed
    {"id": "datasource_2001_2002_2001_1", "hydrant": 12}, # Only hydrants through 12 are committed
  ]
}
```

When a realtime node crashes and starts back up, it would delete any hydrants numbered higher than the ones in the commit file.